src/libraries/System.Private.CoreLib/src/System/Text/Rune.cs

```diff
@@ -1089,7 +1122,11 @@ public static UnicodeCategory GetUnicodeCategory(Rune value)
         private static UnicodeCategory GetUnicodeCategoryNonAscii(Rune value)
         {
             Debug.Assert(!value.IsAscii, "Shouldn't use this non-optimized code path for ASCII characters.");
+#if !NETSTANDARD2_0
             return CharUnicodeInfo.GetUnicodeCategory(value.Value);
+#else
+            return CharUnicodeInfo.GetUnicodeCategory(value.ToString(), 0);
```

Comment:
If this value is within the BMP, we should call the overload `CharUnicodeInfo.GetUnicodeCategory(char)` instead. That will avoid the allocation.

Reply:
Good call. Will fix.